avo-hq/avo
# feature: add visible option to panels

# Description
This PR adds a `visible` option to Panels, which should hide all the contained fields as well as the panel.

This new option shouldn't affect `main_panel` only `panel`

Fixes [#https://github.com/avo-hq/avo/issues/2971](https://github.com/avo-hq/avo/issues/2971)



# Checklist:



## Screenshots & recording



https://github.com/user-attachments/assets/0bfa2d76-6ff3-4f31-80dd-3ab6453ca63c

## Manual review steps


1. Add `visible: -> { false }` to a panel
2. Make sure it hides all the panel header and all fields contained
3. Move set it back to true and they should reappear
4. Ensure it doesn't work when changing the visibility of `main_panel`
5. Ensure it doesn't hide anything when visible isn't set

Manual reviewer: please leave a comment with output from the test if that's the case.
